### PR TITLE
 Update form spacing and add form layout components

### LINKED ISSUE #508: Create a Form component [forms]
We need flexible but consistent forms on our website properties.

These specs are based off of the input fields issue https://github.com/mozilla/protocol/issues/430

Attached are specs + examples of different form types build from all the same atoms
form-specs

form-examples

### PR DESCRIPTION
**Pull request by stephaniehobson**
## Description

Add form heading, header, footer, field, and sets.

- add two demo forms
- form colour is now inherited from mzp-c-form, this makes the dark theme easier to manager
- add dark theme
- remove lists from examples where they were used for form spacing
- rename form-label mixin to field-label

---

- [x] I have documented this change in the design system.
- [x] I have recorded this change in `CHANGELOG.md`.

### Issue

Fix #508

### Testing

The form field, label, and control classes will probably feel like unnecessary abstraction until the new error message and group styles merge. 

Field naming should mirror the naming and nesting used in the choices component #573.

http://localhost:3000/patterns/templates/forms.html
http://localhost:3000/patterns/molecules/forms.html
http://localhost:3000/demos/forms/long.html
http://localhost:3000/demos/forms/accounts.html

### CODE REVIEW

**Comment by stephaniehobson** [comment]
This is ready for review but I'm really unhappy with the documentation and not sure how to fix it. I won't say what's bothering me in case it's only a problem in my head - in which case the reviewer can say the documentation is fine :P

File: src/templates/drizzle/item.hbs [hunk 1]
```diff
@@ -32,24 +32,26 @@
...
   {{else}}
-    {{!-- Bidi controls --}}
-    {{#unless data.hidepreview}}
-    {{#> drizzle.bidi id=(toSlug name)}}
-      {{name}}
-    {{/drizzle.bidi}}
-    {{/unless}}
-    {{!-- Preview --}}
-    <div class="protosite-pattern-embed">
+    {{#unless data.hideembed}}
```
Comment (craigcook) [inline src/templates/drizzle/item.hbs:35]: 🔥

File: src/patterns/molecules/forms/header-footer.hbs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    {{#embed "patterns.molecules.forms.fields.input" id="id_user_email" name="user_email" type="text"  info="true" }}
+      {{#content "label"}}Label{{/content}}
+      {{#content "info"}}Supplementary information about the form fields can be included in a field-info.{{/content}}
+    {{/embed}}
+  </fieldset>
+  <div class="mzp-c-form-footer">
+    <p>This paragraph and the elements below are in a form-footer</p>
+    <button class="mzp-c-button">Button</button>
+    <p class="mzp-c-form-info mzp-t-xs">The button container component is optional - use it for button alignment options if you need them.</p>
+  <div>
```
Comment (craigcook) [inline src/patterns/molecules/forms/header-footer.hbs:32]:
```suggestion
  </div>
```

File: src/pages/demos/forms/accounts.hbs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        <p>Enter your email to access Firefox Accounts.</p>
+      </div>
+      {{#embed "patterns.molecules.forms.fields.input" id="fxa-email-field" type="email" info="true" fieldClass="mzp-l-strech" }}
+        {{#content "label"}}Email address{{/content}}
+        {{#content "info"}}By proceeding, you agree to the <a href="https://accounts.firefox.com/legal/terms">Terms of Service</a> and <a href="https://accounts.firefox.com/legal/privacy">Privacy Notice</a>.{{/content}}
+      {{/embed}}
+      <footer class="mzp-c-form-footer">
+        <div class="mzp-c-button-container mzp-l-strech">
+          <button type="submit" class="mzp-c-button mzp-t-primary mzp-t-product">Continue</button>
+          <p class="mzp-c-button-info">
```
Comment (craigcook) [inline src/pages/demos/forms/accounts.hbs:19]: This class doesn't seem to have any added styling. I'm guessing the text should be small, and maybe a top margin? 
image

Comment (stephaniehobson) [reply src/pages/demos/forms/accounts.hbs:19]: It's covered in #581.

**Comment by stephaniehobson** [comment]
Updated. Let me know if you think we need to do something with the button container styles being in a different PR.

File: src/pages/demos/forms/accounts.hbs (hunk 3 above)
Comment (craigcook) [reply src/pages/demos/forms/accounts.hbs:19]: Ah that explains it.

**Comment by stephaniehobson** [comment]
Merge conflict resolved. It would be great to get this merged today if possible.

**Pull request merged**